TddShop/TddShop.Cli/Shipment/RomanConverter.cs

```diff
@@ -0,0 +1,197 @@
+    //Main part
+    public partial class RomanConverter : IRomanConverter
+    {
+        private Dictionary<int, string> _dictionary;
+        private string _thousandSign;
+        private string _visculumSign = "\u0305";
+        private string _visculumDoubleSign = "\u0305\u0305";
+
+        //TODO:Too big, make it shorter
+        public string Convert(int arabicValue)
+        {
+            bool isVisculum = arabicValue >= 4000 ? true : false;
+            _thousandSign = visculumThousandSign(isVisculum);
+            _dictionary = createDictionary(isVisculum);
+
+            var convertionResult = toRoman(arabicValue);
+            var validator = fromRoman(convertionResult);
+
+            if (validator == arabicValue && validator != 0)
+            {
+                // For less than 4000 change thousandSign to M
+                convertionResult = arabicValue < 4000 ? convertionResult
+                                        .Replace($"{_thousandSign}", "M") :
+                                                    convertionResult;
+                return convertionResult
+                       .Replace($"C{_thousandSign}", "CM");
+            }
+            else
+            {
+                return "error";
+            }
+        }
+
+        private Boolean checkAllowableChars(string romanValue)
+        {
+            string allowableChars = getAllowableChars(_dictionary);
+            return romanValue.Replace($"{_visculumSign}", "")
+                             .All(c => allowableChars.Contains(c));
+        }
+
+        //TODO:Too big, make it shorter
+        public int Convert(string romanValue)
+        {
+            bool isVisculum = true;
+            _thousandSign = visculumThousandSign(isVisculum);
+            _dictionary = createDictionary(isVisculum);
+
+            //Case with "I vinculum M" which is "MM" or "I\u0305I\u0305"
+            romanValue = romanValue.ToUpper().Replace("M", _thousandSign);
+
+            //Check string if contains chars from dictionary
+            bool charsCheck = checkAllowableChars(romanValue);
+
+            var convertionResult = charsCheck ? fromRoman(romanValue) : 0;
+            var validator = toRoman(convertionResult);
+
+            //Validate if conversion is ok
+            if (validator == romanValue && convertionResult != 0)
+            {
+                // For less than 4000 change thousandSign to M
+                romanValue = convertionResult < 4000 ? romanValue
+                                        .Replace($"{_thousandSign}", "M") :
+                                                 romanValue;
+                return convertionResult;
+            }
+            else
+            {
+                return 0;
+            }
+        }
+
+    }
+
+    //Converter internal part
+    partial class RomanConverter
+    {
+        //TODO:Too many nestings, maybe use LINQ
+        private string toRoman(int number)
```

Comment:
Faktycznie, trochę nam zajęło, żeby zrozumieć o co chodzi. Mógłbyś spróbować z LINQ z metodami Aggregate, TakeWhile ale mogłoby to być jeszcze mniej czytelne.